examples/simulators/utility_forces_numer_deriv/CMakeLists.txt

```diff
@@ -35,7 +35,7 @@
 
 cmake_minimum_required(VERSION 3.10)
 
-project(utility_forces_numer_deriv VERSION 1.0.0 LANGUAGES Fortran)
+project(utility_forces_numer_deriv VERSION 1.0.0 LANGUAGES Fortran CXX)
```

Comment:
Why is CXX needed here?

Reply:
I cannot explain why, but without this change execution of the commands from `INSTALL` fails with the following error:
```
...
++ cp -r /home/astuk/kim-api/build/../examples/simulators/utility_forces_numer_deriv /home/astuk/kim-api/build/WORKSPACE/
++ cd /home/astuk/kim-api/build/WORKSPACE
++ cd utility_forces_numer_deriv
++ mkdir build
++ cd build
++ cmake ..
-- Building for: MinGW Makefiles
-- The Fortran compiler identification is GNU 10.1.0
-- Check for working Fortran compiler: C:/msys64/mingw64/bin/gfortran.exe
-- Check for working Fortran compiler: C:/msys64/mingw64/bin/gfortran.exe - works
-- Detecting Fortran compiler ABI info
-- Detecting Fortran compiler ABI info - done
-- Checking whether C:/msys64/mingw64/bin/gfortran.exe supports Fortran 90
-- Checking whether C:/msys64/mingw64/bin/gfortran.exe supports Fortran 90 - yes
-- Configuring done
CMake Error in CMakeLists.txt:
  Cannot use features from non-enabled language CXX

CMake Generate step failed.  Build files cannot be regenerated correctly.
``` 
Adding "CXX" to the language list helped overcome the problem.

Reply:
CMake version is 3.17.3, by the way.

Reply:
I think this is caused by the compile features setting:
https://github.com/openkim/kim-api/blob/491242cb0af43aa1a3b3dbcefbfce07807adce8a/cpp/src/CMakeLists.txt#L89-L100

Any idea what the "right" way to handle this situation is?  I wonder if the CMake way would be to create separate library targets (alias targets?) for each language binding?

Reply:
Apparently, the CMake developers are already aware of the problem and have fixed it in CMake 3.18.0 (too late for us, unfortunately):

https://gitlab.kitware.com/cmake/cmake/-/issues/20438

With the new CMake version, the consuming target (Fortran-only) will ignore any C++-related requirements such as `cxx_std_17` of the consumed target (kim-api) if the C++ language has not been explicitly enabled. 

To tell the truth, I'm not sure either how to deal with the problem in earlier CMake versions. The `kim-api-targets.cmake` file is set in stone once it has been generated and cannot possibly know which languages are enabled by the consuming target. Even injecting a call to [`enable_language(CXX)`](https://cmake.org/cmake/help/latest/command/enable_language.html) into `kim-api-targets.cmake` appears futile because of the scoping rules it is subject to.

I think, for the time being, developers of consuming targets are forced to explicitly enable `CXX` unless they are using CMake 3.18 or above.

Reply:
> Apparently, the CMake developers are already aware of the problem and have fixed it in CMake 3.18.0 (too late for us, unfortunately):
> 
> https://gitlab.kitware.com/cmake/cmake/-/issues/20438
> 
> With the new CMake version, the consuming target (Fortran-only) will ignore any C++-related requirements such as cxx_std_17 of the consumed target (kim-api) if the C++ language has not been explicitly enabled.

Nice find!  Thanks.  However, the fix was NOT merged into 3.18.0 due to the identification of a troublesome corner case.  As of now it seems the issue is still an open problem.

It seems to me the only way to deal with this at the moment would be to define multiple "interface specific" kim-api targets and export each one for use by consuming projects.  However, this seems like it might be overkill; maybe.  

(The worst case scenario, I think, is something like a big Fortran package that supports optional use of the KIM API.  Either they would need to include special logic to only link with KIM-API if specifically requested; Or their build would explicitly depend on a C++ environment being available.  That could be a deal-breaker; certainly I would not be happy in that situation.)

I'll think on this a bit, but probably we'll go with things as you have them.

Reply:
OK; Yes, let's go with it as is except for a minor adjustment which I'll push here nex.